CHANGELOG.md

```diff
@@ -4,14 +4,21 @@
 ### Notes
 
 - The version requires PostgreSQL 12. If you also want to upgrade PostGIS,
-  update PostGIS firstand run ``ALTER EXTENSION postgis UPDATE;`` in every
+  update PostGIS first and run ``ALTER EXTENSION postgis UPDATE;`` in every
   existing database in the cluster that should be upgraded. For docker-compose
   setups this database update is performed automatically if `DB_UPDATE=true` is
   set for the `db` container (watch the Docker output). CATMAID's documentation
   Docker has more information. If a replication setup is in use, the database
   configuration changes for Postgres 12. CATMAID's replication documentation
   explains what needs to be done.
 
+  In Postgres 12, JIT compilation is available to the planner by default. We
+  found that being more conservative with the use of it helped a few common
+  queries (like the tracing data field of view). We set the required minimum
+  cost of JIT use to 1,000,000 in ``postgresql.conf``::
+
+    jit_above_cost = 1000000
```

Comment:
For Docker possibly and I am tuning Postgres there already automatically. But I would like to investigate this a bit more before we would add that setting there. I could only find a few rare cases where this helped yesterday, but remember this had a broader effect when starting to work on this branch a while back. At that time though, I wasn't yet aware of this weird PostGIS 3 problem, so my numbers from back than are likely wrong to some extent. For now the comment on using it a potential optimization is enough I think. Many of the optimizations also really depend on your available resources, database size and workload. And given that vagrant might be mainly used on dev machines with typically smaller resources than servers, I don't think this will have a big effect. Optimizing also the vagrant DB is generally a good idea though and could be done in a similar fashion to the Docker container.